app/Listeners/PrintLabelSubscriber.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace App\Listeners;
+
+use HMS\Repositories\MetaRepository;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use HMS\Repositories\LabelTemplateRepository;
+use App\Events\Labels\LabelPrintEventInterface;
+
+class PrintLabelSubscriber implements ShouldQueue
+{
+    use InteractsWithQueue;
+    /**
+     * printer port.
+     * @var  int
+     */
+    private $port = 9100;
+
+    /**
+     * @var EntityManagerInterface
+     */
+    protected $em;
+
+    /**
+     * @var MetaRepository
+     */
+    protected $metaRepository;
+
+    /**
+     * LabelTemplate Reposistry.
+     * @var LabelTemplateRepository
+     */
+    protected $labelTemplateRepository;
+
+    public function __construct(LabelTemplateRepository $labelTemplateRepository, MetaRepository $metaRepository)
+    {
+        $this->metaRepository = $metaRepository;
+        $this->labelTemplateRepository = $labelTemplateRepository;
+    }
+
+    /**
+     * Handle the event.
+     *
+     * @param  LabelPrintEventInterface $event
+     * @return void
+     */
+    public function handlePrint(LabelPrintEventInterface $event)
+    {
+        for ($i = 0; $i < $event->getCopiesToPrint(); $i++) {
+            $this->printLabel(
+                $event->getTemplateName(),
+                $event->getSubstitutions()
+            );
+            sleep(1);
```

Comment:
its a pause between printing copies so the job is processed on the device, might not be need but not been able to test in real world